app/src/main/java/org/mozilla/fenix/settings/AccountSettingsFragment.kt

```diff
@@ -158,25 +160,30 @@ class AccountSettingsFragment : PreferenceFragmentCompat() {
 
     private fun getChangeListenerForDeviceName(): Preference.OnPreferenceChangeListener {
         return Preference.OnPreferenceChangeListener { _, newValue ->
-            // Optimistically set the device name to what user requested.
-            val deviceNameKey = context!!.getPreferenceKey(R.string.pref_key_sync_device_name)
-            val preferenceDeviceName = findPreference<Preference>(deviceNameKey)
-            preferenceDeviceName?.summary = newValue as String
-
-            // This may fail, and we'll have a disparity in the UI until `updateDeviceName` is called.
-            lifecycleScope.launch(IO) {
-                try {
-                    accountManager.authenticatedAccount()?.let {
-                        it.deviceConstellation().setDeviceNameAsync(newValue)
+            // The network request requires a nonempty string, so don't persist any changes if the user inputs one.
+            if (newValue.toString().trim() == "") {
```

Comment:
Let's opt for `newValue.toString().trim().isEmpty()` here instead of comparing to empty string